drivers/volume/portworx/portworx.go

```diff
@@ -1084,6 +1097,22 @@ func (p *portworx) validatePVForGroupSnap(pvName, groupID string, groupLabels ma
 	return nil
 }
 
+func (p *portworx) ensureNodesDontMatchVersionPrefix(versionRegex *regexp.Regexp) error {
+	result, err := p.clusterManager.Enumerate()
+	if err != nil {
+		return err
```

Comment:
This should be a retryable error, shouldn't mark it as failed in this case.

Reply:
You mean put the VolumeSnapshot in Pending state so that the controller keeps retrying? Is the use case that users will upgrade Portworx to the 1.4 version and then the pending VolumeSnapshot will succeed?

Reply:
No, if the version is old fail it. But if the cluster enumerate returns an error it needs to be retried.

Reply:
done

Reply:
updated as per discussion